client.go

```diff
@@ -921,6 +935,71 @@ func (c *Client) translateKeys(req *pbuf.TranslateKeysRequest, keys []string) ([
 	return idsResp.IDs, nil
 }
 
+func (c *Client) logImport(index, path string, shard uint64, data []byte) {
+	if c.importLogEncoder == nil {
+		return
+	}
+	c.logLock.Lock()
+	go func() {
+		defer c.logLock.Unlock()
+		l := &importLog{
+			Index: index,
+			Path:  path,
+			Shard: shard,
+			Data:  data,
+		}
+		// Encode is actually threadsafe (with gob), but the lock above is
+		// helpful for knowing when we've finished all these goroutines.
+		err := c.importLogEncoder.Encode(l)
+		if err != nil {
+			c.logger.Printf("writing to import log: %v", err)
+		}
+	}()
+}
+
+// ExperimentalReplayImport takes a data stream which was previously recorded by the import
+// logging functionality and imports it to Pilosa. The target cluster need not
+// be of the same size as the original cluster, but it must already have the
+// necessary schema in place. It is an experimental method and may be changed or
+// removed.
+func (c *Client) ExperimentalReplayImport(r io.Reader) error {
+	dec := newImportLogDecoder(r)
+	for {
+		l := &importLog{}
+		err := dec.Decode(l)
+		if err == io.EOF {
+			return nil
+		} else if err != nil {
+			return errors.Wrap(err, "decoding")
+		}
+
+		// regular import doesn't forward to replicas, so we have to get all
+		// the nodes.
+		nodes, err := c.fetchFragmentNodes(l.Index, l.Shard)
+		if err != nil {
+			return errors.Wrap(err, "fetching fragment nodes")
+		}
+
+		if !strings.Contains(l.Path, "/import-roaring/") {
```

Comment:
Would it make sense to have an `isRoaring` field in the `importLog` struct? Checking the path for the type of import is a bit fragile.